src/playsim/p_effect.cpp

```diff
@@ -979,3 +982,273 @@ void P_DisconnectEffect (AActor *actor)
 		p->size = 4;
 	}
 }
+
+//===========================================================================
+// 
+// ZScript Sprite (DZSprite)
+// Concept by Major Cooke
+// Most code borrowed by Actor and particles above
+// 
+//===========================================================================
+
+DZSprite::DZSprite()
```

Comment:
you shouldn't use a C++ constructor for this, but instead a Construct method, because those are what's called by CreateThinker